idaes/power_generation/unit_models/drum1D.py

```diff
@@ -0,0 +1,1171 @@
+        Define constraints which describe the behaviour of the unit model.
+        """
+
+        # Thermal conductivity of drum
+        self.therm_cond_metal = Param(initialize=40, mutable=True)
+
+        # Thermal conductivity of insulation
+        self.therm_cond_insulation = Param(initialize=0.08, mutable=True)
+
+        # Thermal conductivity of air
+        self.therm_cond_air = Param(initialize=0.03, mutable=True)
+
+        # Density of drum material
+        self.dens_metal = Param(initialize=7753, mutable=True)
+
+        # Heat capacity of drum material
+        self.cp_metal = Param(initialize=486, mutable=True)
+
+        # Young modulus
+        self.Young_modulus = Param(initialize=2.07E5, mutable=True)
+
+        # Poisson's ratio
+        self.Poisson_ratio = Param(initialize=0.292, mutable=True)
+
+        # Coefficient of thermal expansion
+        self.coefficient_therm_expansion = Param(initialize=1.4E-5,
+                                                 mutable=True)
+
+        # Constant related to Ra,
+        # (gravity*expansion_coefficient*density
+        # /viscosity/thermal_diffusivity)^(1/6)
+        # Use properties at 50 C and 1 atm, 6.84e7^(1/6)=20.223
+        self.const_Ra_root6 = Param(initialize=20.223, mutable=True)
+
+        # Constant related to Nu for free convection,
+        # 0.387/(1+0.721*Pr^(-9/16))^(8/27)
+        # Use properties at 50 C and 1 atm
+        self.const_Nu = Param(initialize=0.322, mutable=True)
+
+        # Ambient pressure
+        self.pres_amb = Param(initialize=1.0E5, doc="Ambient Pressure")
+
+        # Ambient temperature
+        self.temperature_ambient = Var(self.flowsheet().config.time,
+                                       initialize=298.15,
+                                       doc="Ambient Temperature")
+
+        # Inside heat transfer coefficient
+        self.h_in = Var(self.flowsheet().config.time, initialize=1,
+                        doc="Inside Heat Transfer Coefficient")
+
+        # Outside heat transfer coefficient
+        self.h_out = Var(self.flowsheet().config.time, initialize=1,
+                         doc="Outside Heat Transfer Coefficient")
+
+        # Insulation free convection heat transfer coefficient
+        self.h_free_conv = Var(self.flowsheet().config.time, initialize=1,
+                               doc="Insulation Free Convection"
+                               "Heat Transfer Coefficient")
+
+        # Ra number of free convection
+        self.N_Ra_root6 = Var(self.flowsheet().config.time, initialize=80,
+                              doc="1/6 Power of Ra"
+                              "Number of Free Convection of Air")
+
+        # Nu number  of free convection
+        self.N_Nu = Var(self.flowsheet().config.time, initialize=1,
+                        doc="Nu Number of Free Convection of Air")
+
+        # Define the continuous domains for model
+        self.r = ContinuousSet(bounds=(self.drum_ri, self.drum_ro))
```

Comment:
I would use a more descriptive name here too - "radial_domain" perhaps? (The 1D control volume uses length_domain from memory).